Update Python requirements for Ansible inclusion requirements document (#281)
ansible/ansible-documentation: docs/docsite/rst/community/collection_contributors/collection_requirements.rst

One example scenario where the "even if" clause comes into play is when using cloud modules. These modules mostly run on the controller node but in some environments, the controller might run on one machine inside a demilitarized zone which cannot directly access the cloud machines. The user has to have the cloud modules run on a bastion host/jump server which has access to the cloud machines.

> An **eligible controller Python version** for a collection is a Python version that is supported on the controller side by at least one ansible-core version that the collection supports. Similarly, an **eligible target Python version** for a collection is a Python version that is supported on the target side by at least one ansible-core version that the collection supports. The eligible controller and target Python versions can be determined from the :ref:`ansible_core_support_matrix` and from the ``requires_ansible`` value in ``meta/runtime.yml`` in the collection.

Reviewer: ```suggestion
An **eligible controller Python version** for a collection is a Python version that is supported on the controller side by at least one ansible-core version that the collection supports. Similarly, an **eligible target Python version** for a collection is a Python version that is supported on the target side by at least one ansible-core version that the collection supports. The eligible controller and target Python versions can be determined from the :ref:`ansible_core_support_matrix` and from the ``requires_ansible`` value in ``meta/runtime.yml`` in the collection.
```
- can we replace the word `eligible` with a synonym if possible? `allowed` or `acceptable` maybe? can be easier to understand for not native speakers
- can we add an example? something like `For example, if your collection supports ansible-core version A (supports Python ...) and ansible-core version B (supports Python ...), the acceptable controller Python version is ...`?. Same for target versions.

Author: Hmm, 'acceptable' could be ok, or maybe 'admissible'. I'd like to hear some more comments on this, also from native english speakers :)

About adding an example: there is already one starting directly after the diff that GitHub shows ends, in the part on documentation requirements. Is that one enough, or should we have another one here?

Reviewer 2: I'd agree with the word `acceptable` than  `allowed`

I will also like to suggest that this part of the sentence "An **eligible controller Python version** for a collection is a Python version that is supported on the controller side by at least one ansible-core version that the collection supports." 

could be rewritten as "An **acceptable controller Python version** for collection is a Python version that is supported on the controller side by at least one ansible-core version that the collection supports." 

Similarly, for target versions, take off the `a` before `collection`

Author: Are you sure that removing the `a` before `collection` is fine? That sounds really wrong to me.

Reviewer: about the examples, if there are any, it's OK

Reviewer 3: `acceptable` and/or `desirable` sounds like good options here.

I agree, removing `a` before  `collection` does not sound right.
 
 "An acceptable controller Python version for a collection is the Python version(s) that is supported on the controller side by at least one ansible-core version that the collection supports."

`the Python version(s)` does this change sound more accurate?

Reviewer 4: I'll throw another one into the mix and say `suitable` because it has less negative/weird connotation than `acceptable` and `desirable`. Also `eligible` doesn't quite fit imo. Something can be eligible without really being suitable.

.. _coll_controller_req:

Controller environment
~~~~~~~~~~~~~~~~~~~~~~

> Collections MUST support all eligible controller Python versions in the controller environment, unless required libraries do not support these Python versions. The :ref:`Steering Committee <steering_responsibilities>` can grant other exceptions on a case-by-case basis. 

Reviewer: ```suggestion
Collections MUST support all eligible controller Python versions in the controller environment, unless required libraries do not support these Python versions. The :ref:`Steering Committee <steering_responsibilities>` can grant other exceptions on a case-by-case basis. 
```
i would replace `eligible` also here (see the comment above) and everywhere with a synonym.

Reviewer 5: Could this all be simplified to:
Collections MUST support all Python versions listed in ref`ansible_core_support_matrix` for all ansible-core versions that the collection supports. The exception is if  required libraries for the collection do not support these Python versions. Steering committee blablabla...

Then we avoid having to define 'eligible' python versions entirely have less for the reader to ...read and understand.
`

Author: I think it's too late to make larger changes to this, since we already started voting on it - see https://github.com/ansible-community/community-topics/discussions/268. Right now we should stick so simple wording changes or *small* formulation changes that don't change meanings.

Also if we simplify it, it should be

> Collections MUST support all Python versions listed in :ref:`ansible_core_support_matrix` in the Controller Python column for all ansible-core versions that the collection supports.

Also this needs to be repeated six times in the text (sometimes with Controller Python, sometimes with Target Python). I'm not sure that really improves readability.

Reviewer: let's merge it as voted and then think about simplifications. I think the requirements (especially Python-related) look hard to read and understand overall (at least to me). But let's do it in a dedicated separate PR

> Collections MUST support all eligible controller Python versions in the controller environment, unless required libraries do not support these Python versions. Other exceptions from this rule can be granted on a case-by-case basis by the :ref:`Steering Committee <steering_responsibilities>`.

Reviewer 6: ```suggestion
Collections MUST support all eligible controller Python versions in the controller environment, unless required libraries do not support these Python versions. The :ref:`Steering Committee <steering_responsibilities>` can grant other exceptions on a case-by-case basis.

The collection MUST document all eligible controller Python versions that are not supported in the controller environment. See :ref:`coll_python_docs_req` for details.

Other environment
~~~~~~~~~~~~~~~~~

> Collections MUST support all eligible controller Python versions in the other environment, unless required libraries do not support these Python versions. Other exceptions from this rule can be granted on a case-by-case basis by the :ref:`Steering Committee <steering_responsibilities>`.

Reviewer 6: Let's avoid passive voice.

```suggestion
Collections MUST support all eligible controller Python versions in the other environment, unless required libraries do not support these Python versions. The :ref:`Steering Committee <steering_responsibilities>` can grant other exceptions on a case-by-case basis. 
```
